File: src/main/java/com/kgaft/VulkanLib/Instance/InstanceLogger/DefaultVulkanLoggerCallback.java
package com.kgaft.VulkanLib.Instance.InstanceLogger;

import java.io.PrintStream;
import java.util.Date;

public class DefaultVulkanLoggerCallback implements IVulkanLoggerCallback{
    @Override
    public void messageRaw(int messageSeverity, int messageType, long pCallbackData, long pUserData) {

    }

    @Override
    public void translatedMessage(String severity, String type, String message, boolean isError) {
        PrintStream output = isError?System.err:System.out;
        String outputMessage = new Date()+" VULKAN "+type+" ["+severity+"] "+message;
        output.println(outputMessage);
    }

    @Override
    public VulkanLoggerCallbackType getCallbackType() {
        return VulkanLoggerCallbackType.TRANSLATED_VULKAN_DEFS;
    }
}

File: VulkanLib/InstanceLogger/DefaultVulkanLoggerCallback.cs
using Silk.NET.Vulkan;

namespace VulkanLib.InstanceLogger;

public class DefaultVulkanLoggerCallback : IVulInstanceLoggerCallback
{
    public unsafe void messageRaw(DebugUtilsMessageSeverityFlagsEXT messageSeverity, DebugUtilsMessageTypeFlagsEXT messageTypes,
        DebugUtilsMessengerCallbackDataEXT* pCallbackData, void* pUserData)
    {
        throw new NotImplementedException();
    }

    public void translatedMessage(string severity, string type, string message, bool isError)
    {
        var output = isError?Console.Error:Console.Out;
        String outputMessage = DateTime.Now+" VULKAN "+type+" ["+severity+"] "+message;
        output.WriteLine(outputMessage);
    }

    public VulkanLoggerCallbackType getCallbackType()
    {
        return VulkanLoggerCallbackType.TRANSLATED_VULKAN_DEFS;
    }
}